File: tests/fixtures/edgecase_repo/javascript/src/module3.js
const { Service2 } = require("./module2");

class Base3 {
  constructor(name) {
    this.name = name;
  }
}

class Service3 extends Base3 {
  handle(payload) {
    this.validate(payload);
    return new Service2("x").handle(payload);
  }

  validate(payload) {
    return payload !== null;
  }
}

function helper3(value) {
  return value;
}

module.exports = { Service3 };

File: tests/fixtures/edgecase_repo/rust/src/module3.rs
use crate::module2::Service2;

pub struct Base3 {
    pub name: String,
}

pub struct Service3 {
    pub base: Base3,
}

pub trait Handler3 {
    fn handle(&self, payload: &str) -> bool;
}

impl Handler3 for Service3 {
    fn handle(&self, payload: &str) -> bool {
        self.validate(payload);
        true
    }
}

impl Service3 {
    fn validate(&self, payload: &str) {}
}

pub fn helper_3(value: &str) -> &str {
    value
}
